### PR TITLE
Add error message and tests for `plot_motion`.

### LINKED ISSUE #2725: `plot_motion` does not accept `axes` or `ax` argument
The `widgets.motion.plot_motion()` function asserts that `ax` and `axes` is `None` [here](https://github.com/SpikeInterface/spikeinterface/blob/fa57feeca6b403ae8595204f18313f038f7dc5ac/src/spikeinterface/widgets/motion.py#L76) but the [documentation](https://spikeinterface.readthedocs.io/en/latest/api.html#spikeinterface.widgets.plot_motion) says these should be accepted inputs.

Is there a reason for this? Happy to open a PR to make the function support `ax` / `axes` arguments if not.

Also the latest docs have a number of references to `si.load_motion_info` and `si.plot_motion` but it appears these have since moved, also happy to open PR ASAP but leaving here as don't want to forget.

### PR DESCRIPTION
**Pull request by JoeZiminski**
This PR closes #2725 by adding an error message when `ax` or `axes` arguments are passed, these are `backend_kwargs` which are not supported for this function.

This is slightly suboptimal as the documentation still says they are accepted, but at least it is clear what to do when it does go wrong. I looked into making the function support passing `axes` but it would make things very complicated for (I don't think?) not too much benefit. 

I also added tests for the function and made a couple of minor changes:
- Added a little more explanation to the docstrings..
- renamed the variables `x` and `y` to make slightly more descriptive.
- renamed x-axis label `Times [s]` to `Time [s]`.

I had a couple of questions:
- The `motion_lim` is documented as `Tuple` or `None` but it actually expects an `int`.  I guess this could be changed to accept a `Tuple` similar to `depth_lim`?
- at the moment the widget object `.ax` is set to an empty `Axes` object. Maybe it would be clearer to set this to `None` as it is unused?

Also a more general question, the `matplotlib` and `scipy` imports are not at the top of the file, and I see the tests do not expect `scipy` to be installed. I guess the logic is the same for `matplotlib` which is why it is not top of file? Personally I would add any 'core' python modules to the general dependencies and then there is less maintenance required to have to workaround users potentially not having them installed. I don't think anyone will have trouble installing `matplotlib` or `scipy` unlike heavier modules such as `numba`. However may there are other considerations (e.g. docker images?).

### CODE REVIEW

**Comment by alejoe91** [comment]
> I had a couple of questions:
> 
> * The `motion_lim` is documented as `Tuple` or `None` but it actually expects an `int`.  I guess this could be changed to accept a `Tuple` similar to `depth_lim`?

Yeah I think that sounds super reasonable

> * at the moment the widget object `.ax` is set to an empty `Axes` object. Maybe it would be clearer to set this to `None` as it is unused?

Agreed

> Also a more general question, the `matplotlib` and `scipy` imports are not at the top of the file, and I see the tests do not expect `scipy` to be installed. I guess the logic is the same for `matplotlib` which is why it is not top of file? Personally I would add any 'core' python modules to the general dependencies and then there is less maintenance required to have to workaround users potentially not having them installed. I don't think anyone will have trouble installing `matplotlib` or `scipy` unlike heavier modules such as `numba`. However may there are other considerations (e.g. docker images?).

The idea behind `core` VS `full` is that `core` can be used as a super lightweight module only to use the SI data model, e.g., from a third party package. Adding main Python modules to core could result in dependencies issues for these kind of 3rd party minimal usage.

**Comment by JoeZiminski** [comment]
Thanks @alejoe91! I've made the related changes, this is now ready for review.

Out of interest what kind of third-party apps need low-overhead installations? This would be useful for me to know in general when thinking about dependencies on other packages.

File: src/spikeinterface/widgets/motion.py [hunk 1]
```diff
@@ -7,7 +7,16 @@
...
 class MotionWidget(BaseWidget):
     """
-    Plot unit depths
+    Plot a summary of motion correction outputs. The plots shown are:
+
+    1) 'Peak depth' plot displaying the peak location for all detected spikes over time
+       before motion correction.
+    2) 'Corrected peak depth' displayed the peak location for all detected spikes
+       over time after motion correction.
+    3) 'Motion vectors' plot shows the estimated displacement per channel
```
Comment (JoeZiminski) [inline src/spikeinterface/widgets/motion.py:16]: I actually think I got this wrong, it is displacement across binned depths, not channels? In general I can fix these docstring to better use the terminology from the drift correction tutorial.

Comment (JoeZiminski) [reply src/spikeinterface/widgets/motion.py]: Okay I think I have fixed it now, by replacing "per channel" with "per spatial bin".

**Comment by JoeZiminski** [comment]
Hey @samuelgarcia would you be happy with merging the changes to the `motion.py` in this PR (without the tests)? Assuming the changes to `motion.py` will not make the refactoring harder. If so I can split the changes to `motion.py` and the tests and add the tests back in later after refactoring.

**Comment by JoeZiminski** [comment]
@samuelgarcia  please ignore the above comment, superseded by #3068, and we can add tests later.

**Pull request closed without merging**